uefi/src/mem/aligned_buffer.rs

```diff
@@ -0,0 +1,117 @@
+// SPDX-License-Identifier: MIT OR Apache-2.0
+
+use alloc::alloc::{alloc, dealloc, Layout, LayoutError};
+use core::error::Error;
+use core::fmt;
+use core::ptr::NonNull;
+
+/// Helper class to maintain the lifetime of a memory region allocated with a non-standard alignment.
+/// Facilitates RAII to properly deallocate when lifetime of the object ends.
+///
+/// Note: This uses the global Rust allocator under the hood.
+#[derive(Debug)]
+pub struct AlignedBuffer {
+    ptr: NonNull<u8>,
+    layout: Layout,
+}
+
+impl AlignedBuffer {
+    /// Allocate a new memory region with the requested len and alignment.
+    pub fn from_size_align(len: usize, alignment: usize) -> Result<Self, LayoutError> {
+        let layout = Layout::from_size_align(len, alignment)?;
+        Ok(Self::from_layout(layout))
+    }
+
+    /// Allocate a new memory region with the requested layout.
+    #[must_use]
+    pub fn from_layout(layout: Layout) -> Self {
+        let ptr = unsafe { alloc(layout) };
+        let ptr = NonNull::new(ptr).expect("Allocation failed");
```

Comment:
I'm not sure if it is nice to just panic instead of report an error. Perhaps we need a 

```rust
enum AlignedBufferError {
    Layout(LayoutError),
    OutOfMemory,
}
```

I'd like to hear @nicholasbishop opinion on that